### PR TITLE
fix: endpointIp -> endpointIP

### PR DESCRIPTION
**Pull request by acpana**
Adds resource overrides for `CloudIDSEndpoint` to rename the output only field `endpointIp` to `endpointIP`.

### CODE REVIEW

File: pkg/resourceoverrides/cloudids_overrides.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+			}
+
+			// Rename endpointIp to endpointIP
+			observedState.Properties["endpointIP"] = endpointIp
+			delete(observedState.Properties, "endpointIp")
+
+			return nil
+		},
+		PostActuationTransform: func(original, reconciled *k8s.Resource, tfState *terraform.InstanceState, dclState *unstructured.Unstructured) error {
+			reconciled.Status["endpointIP"] = reconciled.Status["endpointIp"]
```
Comment (maqiuyujoyce) [inline pkg/resourceoverrides/cloudids_overrides.go:60]: Should the logic be:

If `status.endpointIp` exists, add field `status.endpointIP` and remove field `status.endpointIp`?

**Review by maqiuyujoyce — COMMENTED** [review]
Thank you for adding the resource override, @acpana !

Could you add a observed state test to cover it? Context: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2128

File: pkg/resourceoverrides/cloudids_overrides.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+				observedState.Properties["endpointIP"] = endpointIp
+				delete(observedState.Properties, "endpointIp")
+			}
+
+			return nil
+		},
+		PostActuationTransform: func(original, reconciled *k8s.Resource, tfState *terraform.InstanceState, dclState *unstructured.Unstructured) error {
+			endpointIp, ok := reconciled.Status["endpointIp"]
+			if !ok {
+				return fmt.Errorf("endpointIp field was not populated")
```
Comment (maqiuyujoyce) [inline pkg/resourceoverrides/cloudids_overrides.go:64]: Why is it an error? I don't think we require all the output-only field be populated? Or maybe I miss anything?

File: pkg/resourceoverrides/cloudids_overrides.go (hunk 1 above)
Comment (acpana) [reply pkg/resourceoverrides/cloudids_overrides.go]: Yes initially the PR was a wip and I got a chance to polish it up now

**Review by maqiuyujoyce — COMMENTED** [review]
Have you tried to apply this resource locally and verify the output of `kubectl describe` looks as expected? I'm especially curious whether managedFields matches the new field name.

File: pkg/resourceoverrides/cloudids_overrides.go (hunk 2 above)
Comment (acpana) [reply pkg/resourceoverrides/cloudids_overrides.go:64]: that's a good point! I thought we required all output only fields to be populated but it sounds like that's not the case today.

Comment (maqiuyujoyce) [reply pkg/resourceoverrides/cloudids_overrides.go]: AFAIK, output-only fields can be optional in general. We will need to thoroughly understand the API to make a decision.

**Comment by maqiuyujoyce** [comment]
> Could you add a observed state test to cover it? Context: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2128

Thank you for your comment in #2128, @acpana ! As long as there are golden object (that contains the observed state) comparison for this resource in presubmit, I'm good!

File: pkg/resourceoverrides/cloudids_overrides.go (hunk 2 above)
Comment (acpana) [reply pkg/resourceoverrides/cloudids_overrides.go]: `endpointIP` is always present in this case I think. From the integ test `endpointIP` seems to be well populated in the managedFields too.

**Comment by yuwenma** [comment]
/lgtm
/approve

Thank you for the quick fix!

**Pull request merged**